### PR TITLE
Adding new api for storage and catalog to select storage

### PR DESCRIPTION
**Pull request by zmhassan**
# Description

Adding simple storage api and proxy to replace the complexity that exists when running ML and selecting different storage providers.

### CODE REVIEW

**Comment by zmhassan** [comment]
@HumairAK so this is a smaller chunk of a bigger PR to replace the existing storage subsystem within LAD.

File: anomaly_detector/storage/es_storage.py [hunk 1]
```diff
@@ -62,6 +64,36 @@ def _prep_index_name(self, prefix):
...
+    def store_results(self, data):
+        """Store results back to ES."""
+        index_out = self._prep_index_name(self.config.ES_TARGET_INDEX)
+
+        actions = [{"_index": index_out, "_type": "log", "_source": data[i]} for i in range(len(data))]
+
+        helpers.bulk(self.es, actions, chunk_size=int(len(data) / 4) + 1)
+
+
+class ElasticSearchDataSource(StorageSource, DataCleaner):
```
Comment (HumairAK) [inline anomaly_detector/storage/es_storage.py]: I think you meant to also inherit ESStorage

File: anomaly_detector/storage/storage.py [hunk 2]
```diff
@@ -20,6 +20,10 @@ def store_results(self, entries):
         """Store results back to storage backend."""
         raise NotImplementedError("Please implement the <store_results method>")
 
+
+class DataCleaner():
```
Comment (HumairAK) [inline anomaly_detector/storage/storage.py]: Parenthesis here are not needed

File: tests/test_storage_catalog.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    assert len(data) == 512
+
+
+def test_storage_catalog_local_file_save_data(config):
+    """Test anomaly detection training on public dataset."""
+    sc = StorageCatalog(config=config,  storage_api="local.sink")  # config, job_type, storage_api):
+    storage: StorageSink = sc.get_storage_api()
+    sample_output={"Results":"False"}
+    storage.store_results(sample_output)
+    data=None
```
Comment (HumairAK) [inline tests/test_storage_catalog.py]: There are some white space errors here from pep8 `line 27, 28, 35, 37, etc.`

**Comment by zmhassan** [comment]
@HumairAK All issues fixed.

**Pull request merged**